### PR TITLE
Release atlas-xcache (SOFTWARE-3583)

### PR DESCRIPTION
**Pull request by brianhlin**

### CODE REVIEW

File: configs/common-cache/config.d/40-osg-caching-plugin.cfg [hunk 1]
```diff
@@ -5,46 +5,79 @@
...
+ofs.osslib  libXrdPss.so
+# Second, the cache underlying the proxy.
+pss.cachelib libXrdFileCache.so
 
-# The location of data federation this XCache participates in
-#
-# By default, we set this to the OSG Data Federation
-pss.origin redirector.osgstorage.org:1094
+if named atlas-xcache
+   pss.origin =
```
Comment (matyasselmeci) [inline configs/common-cache/config.d/40-osg-caching-plugin.cfg]: Should be a comment here to explain this setting.

File: configs/common-cache/config.d/40-osg-caching-plugin.cfg [hunk 2]
```diff
@@ -5,46 +5,79 @@
...
+   pss.setopt ConnectTimeout 25
+   pss.setopt ConnectionRetry 2
+fi
 
 # Cache plugin parameters
 #
 # These values should be sufficient for most use cases. To change them, override them
 # in a later file such as 90-local.cfg.
 
+if defined ?~XC_RAMSIZE
```
Comment (matyasselmeci) [inline configs/common-cache/config.d/40-osg-caching-plugin.cfg]: Is this correct? Should it be `if defined ?~XC_BLOCKSIZE`? Also, why does the comment above say 'override them in a later file' if they can be set with environment vars?

File: rpm/xcache.spec [hunk 3]
```diff
@@ -80,6 +80,26 @@ Obsoletes: stashcache-cache-server-auth < 1.0.0
...
 
+########################################
+%package -n atlas-xcache
+Summary: The OSG data federation cache server
+
+Requires: %{name} = %{version}
+Requires: wget
+Requires: xrootd-lcmaps >= 1.5.1
+
+Provides: atlas-xcache = %{name}-%{version}
```
Comment (matyasselmeci) [inline rpm/xcache.spec]: This Provides line is redundant.

**Comment by matyasselmeci** [comment]
I think this is fine. Could you test that the cmake changes work?

**Comment by brianhlin** [comment]
Labeled this as `do-not-merge` because we've got to handle all the data disks (https://github.com/slateci/XCache/blob/master/xcache.cfg#L25-L96) which I've done in the container via with a script that autogenerates the config (https://github.com/opensciencegrid/docker-xcache/blob/master/atlas-xcache/image-config.d/10-atlas-xcache.sh#L5-L11).

**Comment by brianhlin** [comment]
Using my dummy dockerized `osg-build`, I was able to actually catch some build errors and was able to fix them to get a successful build!

**Pull request merged**